File: 加密保护/The Enigma Protector v2.33 Build 20110112/Examples/Registration_RuntimeKey/Delphi/test_unit.pas
unit test_unit;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, ExtCtrls;

type
  TForm1 = class(TForm)
    Label1: TLabel;
    Label2: TLabel;
    Timer1: TTimer;
    lbMessages: TListBox;
    eKey: TEdit;
    eName: TEdit;
    Label4: TLabel;
    Label5: TLabel;
    btnRegister: TButton;
    GroupBox1: TGroupBox;
    btnClose: TButton;
    procedure btnCloseClick(Sender: TObject);
    procedure Timer1Timer(Sender: TObject);
    procedure btnRegisterClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
    procedure ShowOurMessage(AMessage : string);
  end;

var
  Form1: TForm1;

implementation

uses enigma_ide;

{$R *.dfm}

procedure TForm1.btnCloseClick(Sender: TObject);
begin
  Timer1.Enabled := false;
  Close;
end;

procedure TForm1.ShowOurMessage(AMessage : string);
begin
  lbMessages.Items.Add(format('[%s] %s', [TimeToStr(Now), AMessage]));
  lbMessages.ItemIndex := lbMessages.Items.Count - 1;
end;

procedure TForm1.Timer1Timer(Sender: TObject);
var
  Total, Left : Cardinal;
begin
  if EP_RegLoadAndCheckKey then
  begin
    if EP_RegKeyRuntime(Total, Left) then
    begin
      ShowOurMessage(format('Valid license, total runtime minute(s) %d, left %d', [Total, Left]));
    end else
    begin
      ShowOurMessage('License is valid, but not runtime limited!');
    end;
  end else
  begin
    ShowOurMessage('License is invalid!');
  end;
end;

procedure TForm1.btnRegisterClick(Sender: TObject);
begin
  if EP_RegCheckAndSaveKey(PAnsiChar(AnsiString(eName.Text)), PAnsiChar(AnsiString(eKey.Text))) then
  begin
    ShowMessage('New license is valid and installed!');
  end else
  begin
    ShowMessage('License is invalid!');
  end;
end;

end.

File: 加密保护/The Enigma Protector v2.70 Build 20110629/Examples/Registration_RuntimeKey/Bcb/test_unit.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "test_unit.h"

// Enigma API ---------------------------------------------------------------
#include "../../../EnigmaSDK/Bcb/enigma_ide.h"
#pragma link "../../../EnigmaSDK/Bcb/enigma_ide.lib"
// --------------------------------------------------------------------------

//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnCloseClick(TObject *Sender)
{
  Timer1->Enabled = false;
  Close();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnRegisterClick(TObject *Sender)
{
  if (EP_RegCheckAndSaveKey(eName->Text.c_str(), eKey->Text.c_str()))
  {
    ShowMessage("New license is valid and installed!");
  } else {
    ShowMessage("License is invalid!");
  }
}

void __fastcall TForm1::ShowOurMessage(AnsiString AMessage)
{
  lbMessages->Items->Add(AnsiString().Format("[%s] %s", ARRAYOFCONST((TimeToStr(Now()), AMessage))));
  lbMessages->ItemIndex = lbMessages->Items->Count - 1;
}

//---------------------------------------------------------------------------
void __fastcall TForm1::Timer1Timer(TObject *Sender)
{
  if (EP_RegLoadAndCheckKey())
  {
    int Total = 0;
    int Left = 0;
    if (EP_RegKeyRuntime(&Total, &Left))
    {
      ShowOurMessage(AnsiString().Format("Valid license, total runtime minute(s) %d, left %d", ARRAYOFCONST((Total, Left))));
    } else {
      ShowOurMessage("License is valid, but not runtime limited!");
    }
  } else {
    ShowOurMessage("License is invalid!");
  }
}
//---------------------------------------------------------------------------

